src/gen.rs

```diff
@@ -41,9 +41,10 @@ pub fn legacy_wrap<'a, G>(gen: G, x: (&'a mut [u8], usize)) -> Result<(&'a mut [
   where G: SerializeFn<io::Cursor<&'a mut [u8]>> {
       let (buf, offset) = x;
       let (buf, offset) = {
-          let mut cursor = io::Cursor::new(buf);
-          cursor.set_position(offset as u64);
-          let cursor = gen(cursor)?;
+          let mut context = WriteContext::new(io::Cursor::new(buf));
+          context.get_mut().set_position(offset as u64);
+          let context = gen(context)?;
+          let (cursor, _) = context.into_inner();
```

Comment:
Why not just creating the Cursor, set_position, then gen_simple here?

Reply:
Because I added `gen_simple()` after I changed that :) Will fix